### PR TITLE
Add homoglyphs rule types

### PR DESCRIPTION
**Pull request by teodor-yanev**
The implications of homoglyphs type of attacks are described in https://github.com/stacklok/minder/issues/2121
It also contains additional explicatory links and useful examples.

### CODE REVIEW

**Review by jhrozek — COMMENTED** [review]
could we add links that describe what kind of attack this prevents? I don't think they are very well known

File: rule-types/github/homoglyphs_invisible_characters.yaml [hunk 1]
```diff
@@ -0,0 +1,23 @@
+version: v1
+type: rule-type
+name: invisible_characters_check
+context:
+  provider: github
+description: Verifies that pull requests do not add any invisible characters
+guidance: >
+  For every pull request submitted to a repository, this rule will check if the
+  pull request adds a new dependency with invisible characters. If it does, the rule will
+  fail and the pull request will be commented on.
```
Comment (JAORMX) [inline rule-types/github/homoglyphs_invisible_characters.yaml:10]: The guidance is meant to tell you what to do if the rule fails. This is more appropriate for a description.

File: rule-types/github/homoglyphs_mixed_scripts.yaml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+version: v1
+type: rule-type
+name: mixed_scripts_check
+context:
+  provider: github
+description: Verifies that pull requests do not add any text with mixed scripts
+guidance: >
+  For every pull request submitted to a repository, this rule will check if the
+  pull request adds text with mixed scripts. If it does, the rule will
+  fail and the pull request will be commented on.
```
Comment (JAORMX) [inline rule-types/github/homoglyphs_mixed_scripts.yaml:10]: The guidance is meant to tell you what to do if the rule fails. This is more appropriate for a description.

**Review by JAORMX — CHANGES_REQUESTED** [review]

File: rule-types/github/homoglyphs_mixed_scripts.yaml (hunk 2 above)
Comment (teodor-yanev) [reply rule-types/github/homoglyphs_mixed_scripts.yaml]: Thanks: updated

**Review by JAORMX — APPROVED** [review]

**Pull request merged**